src/offcanvas/offcanvas-stack.ts

```diff
@@ -0,0 +1,201 @@
+
+    if (!containerEl) {
+      throw new Error(`The specified offcanvas container "${options.container || 'body'}" was not found in the DOM.`);
+    }
+
+    if (!options.scroll) {
+      this._hideScrollBar();
+    }
+
+    const activeOffcanvas = new NgbActiveOffcanvas();
+    const contentRef =
+        this._getContentRef(moduleCFR, options.injector || contentInjector, content, activeOffcanvas, options);
+
+    let backdropCmptRef: ComponentRef<NgbOffcanvasBackdrop>| undefined =
+        options.backdrop !== false ? this._attachBackdrop(moduleCFR, containerEl) : undefined;
+    let panelCmptRef: ComponentRef<NgbOffcanvasPanel> = this._attachWindowComponent(moduleCFR, containerEl, contentRef);
+    let ngbOffcanvasRef: NgbOffcanvasRef =
+        new NgbOffcanvasRef(panelCmptRef, contentRef, backdropCmptRef, options.beforeDismiss);
+
+    this._registerOffcanvasRef(ngbOffcanvasRef);
+    this._registerPanelCmpt(panelCmptRef);
+    ngbOffcanvasRef.hidden.pipe(finalize(() => this._restoreScrollBar())).subscribe();
+    activeOffcanvas.close = (result: any) => { ngbOffcanvasRef.close(result); };
+    activeOffcanvas.dismiss = (reason: any) => { ngbOffcanvasRef.dismiss(reason); };
+
+    this._applyPanelOptions(panelCmptRef.instance, options);
+
+    if (backdropCmptRef && backdropCmptRef.instance) {
+      this._applyBackdropOptions(backdropCmptRef.instance, options);
+      backdropCmptRef.changeDetectorRef.detectChanges();
+    }
+    panelCmptRef.changeDetectorRef.detectChanges();
+    return ngbOffcanvasRef;
+  }
+
+  get activeInstance() { return this._activeInstance; }
+
+  dismiss(reason?: any) { this._offcanvasRef ?.dismiss(reason); }
+
+  hasOpenOffcanvas(): boolean { return !!this._offcanvasRef; }
+
+  private _attachBackdrop(moduleCFR: ComponentFactoryResolver, containerEl: any): ComponentRef<NgbOffcanvasBackdrop> {
+    let backdropFactory = moduleCFR.resolveComponentFactory(NgbOffcanvasBackdrop);
+    let backdropCmptRef = backdropFactory.create(this._injector);
+    this._applicationRef.attachView(backdropCmptRef.hostView);
+    containerEl.appendChild(backdropCmptRef.location.nativeElement);
+    return backdropCmptRef;
+  }
+
+  private _attachWindowComponent(moduleCFR: ComponentFactoryResolver, containerEl: any, contentRef: any):
+      ComponentRef<NgbOffcanvasPanel> {
+    let panelFactory = moduleCFR.resolveComponentFactory(NgbOffcanvasPanel);
+    let panelCmptRef = panelFactory.create(this._injector, contentRef.nodes);
+    this._applicationRef.attachView(panelCmptRef.hostView);
+    containerEl.appendChild(panelCmptRef.location.nativeElement);
+    return panelCmptRef;
+  }
+
+  private _applyPanelOptions(windowInstance: NgbOffcanvasPanel, options: NgbOffcanvasOptions): void {
+    this._panelAttributes.forEach((optionName: string) => {
+      if (isDefined(options[optionName])) {
+        windowInstance[optionName] = options[optionName];
+      }
+    });
```

Comment:
```suggestion
    this._panelAttributes.forEach((optionName) => windowInstance[optionName] ??= options[optionName]);
```
this should work same (skipping `isDefined`)

Reply:
I don't think so. The existing code modifies the `windowInstance` option only if the corresponding `options` option is defined. Your code would modify the `windowInstance` option only if the `windowInstance option` is nullish, and whatever the value of the `options` option is. (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment)

Reply:
Oh, you're correct. I misread the purpose here